service-proxy.js

```diff
@@ -832,6 +832,8 @@ function reapPeers() {
         }
 
         self.channel.peers.delete(hostPort);
+        delete self.peersToReap[hostPort];
+        delete self.connectedPeers[hostPort];
```

Comment:
This should be unhelpful, because the peer cannot be reaped on the subsequent reaping. This introduces a leak.